REPL: server management commands

daveyarwood opened the pull request:
Closes #13.

This adds 6 new REPL commands, each equivalent to their CLI counterparts:

* `:status` - display the status of the server
* `:list` - list running Alda processes
* `:version` - display Alda client + server versions
* `:down` - stop the server
* `:up` - start the server
* `:downup` - restart the server

This PR also includes various aesthetic adjustments -- see the individual commits for details.

daveyarwood committed:
help text: separate columns with spaces instead of tabs

daveyarwood committed:
un-indent doc details / include a final newline

daveyarwood committed:
new REPL commands: down, downup, list, status, up, version

daveyarwood committed:
sort list of available commands when you run :help

daveyarwood committed:
fix whitespace

daveyarwood commented on src/alda/AldaServer.java:
```diff
@@ -233,6 +235,12 @@ public void down() throws NoResponseException {
     throws NoResponseException, InvalidOptionsException {
     down();
+    try {
+      Thread.sleep(PAUSE_BEFORE_RESTARTING_SERVER);
```
This is to avoid a race condition where you stop the server, then try to start it again before the server has stopped, resulting in the error "There is already a server trying to start on that port."

I guess ideally we would check to confirm that the server is down before we restart it, but I think this is OK for now.

jgkamat commented on src/alda/repl/commands/ReplDown.java:
```diff
@@ -0,0 +1,31 @@
+import jline.console.ConsoleReader;
+
+public class ReplDown implements ReplCommand {
+  public ReplDown() {}
```
These empty constructors aren't required, but I guess it doesn't hurt to have them for readability

daveyarwood replied:
Ah, I didn't know that. I will remove them.

jgkamat commented on src/alda/repl/commands/ReplHelp.java:
```diff
@@ -29,13 +33,33 @@ public void act(String args, StringBuffer history, AldaServer server,
+          String spacing = String.join("", spaces);
+          System.out.println("    :" + key + spacing +
+                             cmd.docSummary() +
+                             (cmd.docDetails() != "" ? " (*)" : ""));
```
This should be `.equals("")` or `.length() == 0` or `isEmpty()`... *sigh*

jgkamat reviewed (commented):
Got a few nitpicks, but overall this is looking really good! :smile: 

:+1: on the logic for sorting/printing the help menu by the way :)

daveyarwood committed:
minor adjustments

- Don't use != to compare strings. In this case, we want to check if the string
  is empty, so we can use isEmpty() for that. isEmpty can trigger a NPE if the
  string is null, so we can check for null too. I wrapped this up in a
  hasDocDetails() helper method.

- Remove empty constructors.

jgkamat reviewed (approved):
LGTM! 😄
